tools/operator/launcher/k8s/network.go

```diff
@@ -2,241 +2,522 @@ package k8s
+	coreConfig.Peer.ID = name
+	coreConfig.Peer.Gossip.ExternalEndpoint = fmt.Sprintf("%s:%d", name, port)
+	coreConfig.Peer.Address = fmt.Sprintf("%s:%d", name, port)
+	coreConfig.Peer.LocalMSPID = mspID
+	coreConfig.Peer.Gossip.Bootstrap = fmt.Sprintf("%s:%d", name, port)
+	coreConfig.Operations.ListenAddress = fmt.Sprintf(":%d", metricsPort)
+	d, err := yaml.Marshal(&coreConfig)
+	if err != nil {
+		return err
+	}
+	cryptoConfigPath := paths.CryptoConfigDir(k8s.Config.ArtifactsLocation)
+	path := paths.JoinPath(cryptoConfigPath, fmt.Sprintf("peerOrganizations/%s/peers/%s.%s", orgName, name, orgName))
+	inputPath := paths.JoinPath(path, fmt.Sprintf("core-%s.yaml", name))
+	err = ioutil.WriteFile(inputPath, d, 0644)
 	if err != nil {
 		return err
 	}
+
 	return nil
 }
 
-//LaunchK8sNetwork - to launch the kubernates components
-func (k K8s) LaunchK8sNetwork(config networkspec.Config, kubeConfigPath string) error {
+func (k8s K8s) generateOrdererConfig(name, orgName, mspID string, port int32, metricsPort int32, ordererConfig fabricconfig.Orderer) error {
 
-	path := paths.ChannelArtifactsDir(config.ArtifactsLocation)
-	inputArgs := []string{paths.JoinPath(path, "genesis.block")}
-	k = K8s{KubeConfigPath: kubeConfigPath, Action: "create", Arguments: inputArgs}
-	_, err := networkclient.ExecuteK8sCommand(k.ConfigMapsNSecretsArgs("genesisblock", "secret"), true)
+	ordererConfig.General.LocalMSPID = mspID
+	var rootCAs []string
+	rootCA := fmt.Sprintf("/etc/hyperledger/fabric/artifacts/msp/tlscacerts/tlsca.%s-cert.pem", orgName)
+	ordererConfig.General.TLS.RootCAs = append(rootCAs, rootCA)
+	ordererConfig.General.ListenPort = int(port)
+	ordererConfig.Operations.ListenAddress = fmt.Sprintf(":%d", metricsPort)
+	d, err := yaml.Marshal(&ordererConfig)
 	if err != nil {
 		return err
 	}
-	k8sServicesFile := paths.ConfigFilePath("services")
-	k8sPodsFile := paths.ConfigFilePath("pods")
-	inputPaths := []string{k8sServicesFile, k8sPodsFile}
-	if config.K8s.DataPersistence == "true" {
-		k8sPvcFile := paths.ConfigFilePath("pvc")
-		inputPaths = append(inputPaths, k8sPvcFile)
-	}
-	k = K8s{Action: "apply", Arguments: inputPaths, KubeConfigPath: kubeConfigPath}
-	_, err = networkclient.ExecuteK8sCommand(k.Args(), true)
+
+	cryptoConfigPath := paths.CryptoConfigDir(k8s.Config.ArtifactsLocation)
+	path := paths.JoinPath(cryptoConfigPath, fmt.Sprintf("ordererOrganizations/%s/orderers/%s.%s", orgName, name, orgName))
+	inputPath := paths.JoinPath(path, fmt.Sprintf("orderer-%s.yaml", name))
+	err = ioutil.WriteFile(inputPath, d, 0644)
 	if err != nil {
-		logger.ERROR("Failed to launch the fabric k8s components")
 		return err
 	}
+
 	return nil
 }
 
-//DownK8sNetwork - To tear down the kubernates network
-func (k K8s) DownK8sNetwork(kubeConfigPath string, config networkspec.Config) error {
+func (k8s K8s) core(nsConfig networkspec.Config) (fabricconfig.Core, error) {
```

Comment:
Can we push this up into the `fabricconfig/core.go`? Or is there something that would keep us from putting it there?

Reply:
yes agree. moved to core.go.